### ISSUE TITLE
Headers on RequestOptions in constructor HttpClient not supported

### ISSUE THREAD

**Issue by dries-2002** [Microsoft/typed-rest-client#65]
### Environment
typed-rest-client version: 1.0.7

### Issue Description
I can instantiate the HttpClient with default headers as follows: 

this.httpClient = new HttpClient("myUserAgent", [basicHandler], {headers: {"Accept": "application/json"}});

But it doesn't send these headers to the endpoint. In fact, it doesn't seem to use them at all. 

When I send the Accept header as additionalHeader on the get method, everything works as expected
response = await this.httpClient.get("https://example.com/clients", { "Accept": "application/json"});

### Expected behaviour
The default headers, defined in the constructor, should be merged with the additional headers sent along with the get/post/... method.

### Actual behaviour
No default headers are sent to the endpoint.

**Comment by stephenmichaelf** [comment]
Thanks for pointing this out @dries-2002. I checked the code and it looks like we are indeed not passing the headers from IRequestOptions through the requests. I will make a fix for this.

You are unblocked since you can pass them in each request, correct?

**Comment by dries-2002** [comment]
Correct.

**Comment by nize** [comment]
I have been trouble-shooting for an unreasonable amount of time too due to this bug :). Is there any plan to fix it?

**Comment by bryanmacfarlane** [comment]
Yes, we will fix shortly.  We also take PRs but let us know soon if you're interested.
